### ISSUE TITLE
MP LAN/HOST session doesn't startup fully

### ISSUE THREAD

**Issue by JohnnyDeeee** [official-antistasi-community/A3-Antistasi#814]
*Version:* (unstable) but happend in 2.2.1 aswell

*Mods:* CBA_A3, RHSAFRF, RHSUSAF, ace, CUP Terrains - Core, RHSGREF, CUP ACE3 Compatibility Addon - Terrains, task_force_radio, CUP Terrains - Maps, GRAD Trenches, BackpackOnChest, ACE Compat - RHS- GREF, ACE Compat - RHS United States Armed Forces, ACE Compat - RHS Armed Forces of the Russian Federation, JSRS SOUNDMOD, Blastcore Edited (standalone version), Enhanced Movement

*Environment*: MP LAN, MP host

*.rpt attatched?* NO

*have you edited the missionfile?*: NO

***Issue:***
After starting the (chernarus_summer) mission (from the editor) as an MP LAN (or MP HOST), you will receive the following error:

```sqf
Error in expression <petros == independent) then {
_varName + serverID + campaignID + "Antistasi" + >
Error position: <+ serverID + campaignID + "Antistasi" + >
Error Generic error in expression
File Antistasi-Chernarus-2-2-2.chernarus_summer\statSave\saveFuncs.sqf..., line 9
```

It seems like we are trying to add integers and strings together and SQF doesn't convert the integers to strings automatically.

After the error, Antistasi won't finish initialising, and thus can't be played...

**Comment by JohnnyDeeee** [comment]
I created a PR that makes sure every variable is converted to a string using the `format` function

**Comment by Spoffy** [comment]
Thanks for submitting the PR, I'll have a look at it properly tomorrow. Out of interest, which variable in the concatenation was erroring? A quick look over the code suggests they should all be strings.

**Comment by JohnnyDeeee** [comment]
Not sure which one, will check that tomorrow

**Comment by jaj22** [comment]
Antistasi certainly writes them to the profile as strings. I did once cause this error during a manual savegame recovery by writing ss_campaignID manually, but that would be user error.

I don't know if other/older versions of Antistasi use integers, but the savegame likely wouldn't be compatible in that case anyway.

**Comment by MeltedPixel** [comment]
While I don't think this was a problem, as I am working on save states right now and can confirm they're all strings however, this isn't a bad idea as it sanitizes the data and makes sure its saved as a string. We can use underscores to separate the elements for readability.

**Comment by JohnnyDeeee** [comment]
Okay but how can you explain the error i'm getting in the following scenario:

1. Start Arma 3
1. Go to editor
1. Open Chernarus summer (not sure if it happens with all maps)
1. Open the `Antistasi-Chernarus-2-2-2.chernarus_summer` (or 2.2.1) mission
1. Hit Play -> Player in Multiplayer (MP)
1. Choose Independent -> Default Commander
1. Hit OK
1. Notice the error on screen (if you started arma with `-showScriptErrors`, else refer to error in issue description)

![afbeelding](https://user-images.githubusercontent.com/9005496/75143003-80bb7800-56f4-11ea-9556-c0f9593e3ad3.png)

Hitting "Continue" to finish the briefing will never further initialise Antistasi. You won't get the option to place your HQ on the map and/or load a previous save...

---

It seems like `serverID` contains a Number in this case

![afbeelding](https://user-images.githubusercontent.com/9005496/75143165-d8f27a00-56f4-11ea-9a61-9d6cabcf35f1.png)

**Comment by JohnnyDeeee** [comment]
The following line finds a serverID which is saved as a Number

https://github.com/official-antistasi-community/A3-Antistasi/blob/970f8739cfd112bf4ccd6bd454d0f53d8b633ad4/A3-Antistasi/functions/init/fn_initServer.sqf#L15

Could it be that this is somehow corrupt or saved from an older version of Antistasi?
That could explain why i'm the only one experiencing this issue...

I'll try to debug this with a brand new profile to see if it happens aswell...

**Comment by JohnnyDeeee** [comment]
I can confirm this issue does not happen when using a new profile.
That means my profile somehow had a Number saved as `ss_ServerID`.

Still like @MeltedPixel mentioned, it is not a bad idea to sanitize the data to make sure everything **can** be saved.

**Comment by Bob-Murphy** [comment]
Spoffy
Please confirm that this can be closed as #815 was manually merged

**Comment by Spoffy** [comment]
Closing now, all resolved.
